### PR TITLE
Add PSK support

### PR DESCRIPTION
**Pull request by gesslerpd**
Depends on pyca/cryptography#4084.

I have seen PSK compatibility in some http://coap.technology/ implementations.

If anything, this PR can be used as a patch for pyOpenSSL users that would like to have PSK compatibility.

### CODE REVIEW

**Comment by gesslerpd** [comment]
I had to edit the CI configuration for the CI to pass for the time being. Once the dependent `cryptography` PR code updates are delivered to a PyPI release, these CI edits can be reverted, except for the removal of python 2.6 support.

**Comment by gesslerpd** [comment]
cc: @alex @reaperhulk 

I can close this PR if PSK support is an unwanted addition to the pyOpenSSL API.
I put up this PR for visibility because I had already done the work to implement it.

Please let me know what should be done with this PR going forward.

Thanks!

**Comment by jlaine** [comment]
This needs to be rebased to remove changes to setup.py and tox.ini : the minimum cryptography version has been raised and py26 removed in master

File: tox.ini [hunk 1]
```diff
@@ -4,6 +4,7 @@ envlist = {pypy,pypy3,py27,py34,py35,py36}{,-cryptographyMaster,-cryptographyMin
 [testenv]
 whitelist_externals =
     openssl
+install_command = pip install --process-dependency-links {opts} {packages}
```
Comment (jlaine) [inline tox.ini]: Is this needed?

**Comment by dangtrinhnt** [comment]
Great work!

**Pull request closed without merging**